### ISSUE TITLE
When user owns a `TMaster`, there's no appropiate state / view before joining the community

### ISSUE THREAD

**Issue by noeliaSD** [status-im/status-desktop#14085]
### Description
Here there are 2 different scenarios shared in the following video to show how now the application behaves when a user that owns a `TMaster` sees the `Request to join` process:

https://github.com/status-im/status-desktop/assets/97019400/308ddd05-2613-4207-945a-42dc0f6e5c51

1. There's is the need of a proper design to indicate that the user is able to request to join since she/he meets the requirements. 
2. Work on the design created for this specific flow.

**Comment by noeliaSD** [comment]
@benjthayer here it's needed a proper design.
@jrainville once the design is done, probably a task for your team.

**Comment by benjthayer** [comment]
> @benjthayer here it's needed a proper design. @jrainville once the design is done, probably a task for your team.

Thanks @noeliaSD good spot. In the instance where the user is looking to join a community, and they hodl the TokenMaster token for that community, I think they would see the TokenMaster token as one of the token options. 

So in the case of C2 test for example, the token requirements would be shown as 1 [TokenMaster token name] OR 0.2 ETH. The TokenMaster token requirement would be shown as met given the user hodls the token and they would be able to proceed to share addresses to join. 

Would that flow cover things?

I think there may also be a couple of bugs captured in the video also - it's not clear why the token gating screen flips into the `Share your chatkey` screen - I think it should stay on the token requirements even if they not met - the join button just remains disabled. Similarly at the end of the video where the token requirements switch to the message `Sorry you don't hodl the necessary tokens...` - again the screen should stay on the token requirements I believe.

<img width="1079" alt="Screenshot 2024-03-21 at 3 49 57 PM" src="https://github.com/status-im/status-desktop/assets/110033914/bb67068c-d5cd-4f50-9efe-f3901ed6026a">

**Comment by jrainville** [comment]
> In the instance where the user is looking to join a community, and they hodl the TokenMaster token for that community, I think they would see the TokenMaster token as one of the token options.

> So in the case of C2 test for example, the token requirements would be shown as 1 [TokenMaster token name] OR 0.2 ETH. The TokenMaster token requirement would be shown as met given the user hodls the token and they would be able to proceed to share addresses to join.

> Would that flow cover things?

Yes that was my idea too.

**Comment by benjthayer** [comment]
> > In the instance where the user is looking to join a community, and they hodl the TokenMaster token for that community, I think they would see the TokenMaster token as one of the token options.
> 
> > So in the case of C2 test for example, the token requirements would be shown as 1 [TokenMaster token name] OR 0.2 ETH. The TokenMaster token requirement would be shown as met given the user hodls the token and they would be able to proceed to share addresses to join.
> 
> > Would that flow cover things?
> 
> Yes that was my idea too.

Ok great thanks @jrainville - am I correct in thinking nothing is needed from a design perspective actually in that case? It would just be this screen but with the 15 UNI replaced with the TM token.

https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=2314-257946&mode=design&t=nwYHKFZVMbaBA0Nu-4
<img width="1277" alt="Screenshot 2024-03-22 at 1 31 24 PM" src="https://github.com/status-im/status-desktop/assets/110033914/841fe165-2704-4584-b3c4-7cdc6bafe56a">

**Comment by jrainville** [comment]
> Ok great thanks @jrainville - am I correct in thinking nothing is needed from a design perspective actually in that case? It would just be this screen but with the 15 UNI replaced with the TM token.

@benjthayer Yeah, that would be fine indeed. It was what I had in mind too, but we wanted to double check.

Thanks!
